sporkfish/searcher/minimax.py

```diff
@@ -77,6 +79,8 @@ def __init__(
             == MoveOrderMode.COMPOSITE
             else None
         )
+        if self._searcher_config.move_order_config.move_order_mode == MoveOrderMode.KILLER_MOVE or self._searcher_config.move_order_config.move_order_mode == MoveOrderMode.COMPOSITE:
+            self._history_table: Dict[chess.Move, int]= dict()
 
```

Comment:
This is the part i am least sure about as i tried to use the similar method for killer as 
`self._history_table = (
           self._history_table =(
            dict()
             if self._searcher_config.move_order_config.move_order_mode
            == MoveOrderMode.HISTORY
            or self._searcher_config.move_order_config.move_order_mode
            == MoveOrderMode.COMPOSITE
            else None
        )`
but kept getting error from mypy.

Reply:
Is this resolved now?

Reply:
<img width="886" alt="Screenshot 2024-03-08 at 19 28 06" src="https://github.com/KYLChiu/sporkfish/assets/47755439/b22518ac-d564-4955-a7e0-71813ef6bbbc">
 
hope... it has errorcode from mypy like this...

Reply:
Two things to do here imo:
* Copy the format (self._history_table = dict() if your_current_statement else None)
* Add `#type: ignore` on the same line for mypy to be happy.

Reply:
why does killer_move not cause a problem with mypy??

Reply:
Hmmm but killer did not have Optional?

Reply:
Killer doesn't have an explicit type hint!